### ISSUE TITLE
Prep for jQuery UI Datepicker rewrite

### ISSUE THREAD

**Issue by scottgonzalez** [KidSysco/jquery-ui-month-picker#75]
Datepicker is undergoing a complete rewrite with a new API. You can see the new API and some notes on the [planning wiki](http://wiki.jqueryui.com/w/page/12137778/Datepicker). You can also test out the code in the [datepicker branch](https://github.com/jquery/jquery-ui/tree/datepicker).

Let us know if you have any questions or if there are any extension points that would make it easier to implement the month picker.

**Comment by benjamin-albert** [comment]
@scottgonzalez Cool! I'm exited to take a look!

**Comment by KidSysco** [comment]
Agreed, thanks for the heads up and we appreciate the offer for support!

**Comment by KidSysco** [comment]
Hi @scottgonzalez,

I thought about it quite a bit last night and I noticed that you are looking to support the Month Input type with the new DatePicker. I mean no disrespect because I have been following your work for years, but the reason this repo exists, is because we felt the functionality is missing from the Date Picker.

It is not a huge need, granted, but there is a small steady steam of folks that show up here looking for month picker support.

I also noticed how you are looking to use the Widget Factory too, which is what the Month Picker is based on. That would make all of this very easy to integrate.

You are the man Scott, so it feels like we should be asking you what you need from us.

If you are not interested in including Month Picker functionality in the new Date Picker, no problem, we can keep doing our thing here. But it is a thought I wanted to throw out there while you are here.

Let us know if there is anything we can do for you!

**Comment by KidSysco** [comment]
Also, if you get a chance, take a look at the HTML 5 Month Picker support built into Google Chrome. I just checked and it is still the same.

https://github.com/KidSysco/jquery-ui-month-picker/wiki/HTML-5-Month-Input-Support

This seems like something that might be interesting to you.

**Comment by scottgonzalez** [comment]
We discussed this in the jQuery UI meeting today with @benjamin-albert. We don't plan on building this support directly into the datepicker widget, but we would like to make it possible to support the various HTML5 date types via extensions. After today's discussion, it's still not clear whether this plugin (with its current design) would benefit from being an extension rather than just a new widget.

We're always open to collaboration and feedback from other developers. We do our best to remove pain points for anyone building on top of jQuery UI, so if you run into any issues or see something that could be easier with a change to jQuery UI itself, just let us know and we'll work with you to figure out the best solution.
